### PR TITLE
Refactor Stress Master Bench

### PR DESCRIPTION
**Pull request by elega**
### What changes are proposed in this pull request?

Extract some variables/functions from MasterBenchParameters into a new created MasterStressBenchBaseParameters class
Extract some variables/functions from MasterBenchTaskResult into a new created MasterBenchTaskResultBase class
Extract some variables/functions from StressMasterBench into a new created StressMasterBenchBase class

### Why are the changes needed?

We are going to introduce a new type of MasterStressBench which allows running more than one operations (e.g. ListDir + CreateFile) at a certain ratio (e.g. 70% Read + 30% Write). To DRY the code, we refactor stress master bench related code first so that some code can be reused by the upcoming MultiOperationStressMasterBench.

### Does this PR introduce any user facing changes?

N/A

### CODE REVIEW

File: stress/shell/src/main/java/alluxio/stress/cli/StressMasterBenchBase.java [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+      }
+    }
+
+    public synchronized T getResult() {
+      return mResult;
+    }
+  }
+
+  @SuppressFBWarnings("BC_UNCONFIRMED_CAST")
+  protected void applyNativeOperation(
```
Comment (dbw9580) [inline stress/shell/src/main/java/alluxio/stress/cli/StressMasterBenchBase.java:318]: Make this a class just like the other classes for HDFS and POSIX interfaces?

File: stress/common/src/main/java/alluxio/stress/master/MasterStressBenchBaseParameters.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+package alluxio.stress.master;
+
+import com.beust.jcommander.Parameter;
+
+/**
+ * This holds all the parameters. All fields are public for easier json ser/de without all the
+ * getters and setters.
+ */
+public abstract class MasterStressBenchBaseParameters extends MasterBenchBaseParameters {
+  public static final String DURATION_OPTION_NAME = "--duration";
```
Comment (dbw9580) [inline stress/common/src/main/java/alluxio/stress/master/MasterStressBenchBaseParameters.java:21]: why move duration into its own class?

**Review by dbw9580 — COMMENTED** [review]
Mostly looks good to me, with a few questions

File: stress/shell/src/main/java/alluxio/stress/cli/AbstractStressBench.java [hunk 3]
```diff
@@ -34,10 +34,15 @@
   @ParametersDelegate
   protected P mParameters;
 
+  /**
+   * Validates the parameters.
+   */
+  public void validateParams() throws Exception {}
```
Comment (dbw9580) [inline stress/shell/src/main/java/alluxio/stress/cli/AbstractStressBench.java:40]: make this abstract?

**Review by dbw9580 — APPROVED** [review]
LGTM

File: stress/shell/src/main/java/alluxio/stress/cli/AbstractStressBench.java (hunk 3 above)
Comment (elega) [reply stress/shell/src/main/java/alluxio/stress/cli/AbstractStressBench.java:40]: i can follow up in my second PR

**Comment by yyongycy** [comment]
alluxio-bot, merge this please

**Pull request merged**